hs_geographic_feature_resource/models.py

```diff
@@ -0,0 +1,390 @@
+        ori_file_info = OriginalFileInfo.objects.get(id=element_id)
+        if ori_file_info:
+            ori_file_info.delete()
+        else:
+            raise ObjectDoesNotExist("No OriginalFileInfo element exists for id: %d."%element_id)
+
+#
+# Define original spatial coverage metadata info
+class OriginalCoverage(AbstractMetaDataElement):
+
+    term = 'OriginalCoverage'
+
+    #original extent
+    #_extent = models.CharField(max_length=1024, null=False, blank=False)
+    northlimit = models.FloatField(null=False, blank=False)
+    southlimit = models.FloatField(null=False, blank=False)
+    westlimit = models.FloatField(null=False, blank=False)
+    eastlimit = models.FloatField(null=False, blank=False)
+
+    #eg., prj string
+    projection_string = models.TextField(max_length=1024, null=True, blank=True)
+    projection_name = models.TextField(max_length=256, null=True, blank=True)
+    datum = models.TextField(max_length=256, null=True, blank=True)
+    unit = models.TextField(max_length=256, null=True, blank=True)
+
+    class Meta:
+        # OriginalCoverage element is not repeatable
+        unique_together = ("content_type", "object_id")
+
+
+    @classmethod
+    def create(cls, **kwargs):
+
+        for limit in ['northlimit', 'southlimit', 'westlimit', 'eastlimit']:
+            if not limit in kwargs:
+                raise ValidationError("For original coverage meta, one or more bounding box limits is missing.")
+
+        metadata_obj = kwargs['content_object']
+        ori_cov = OriginalCoverage.objects.create(northlimit=kwargs['northlimit'], southlimit=kwargs['southlimit'],
+                                                  westlimit=kwargs['westlimit'],eastlimit=kwargs['eastlimit'],
+                                                  content_object=metadata_obj,)
+
+        # # update projection string info
+        for key, value in kwargs.iteritems():
+            if key in ('projection_string','projection_name','datum','unit'):
+                setattr(ori_cov, key, value)
+                ori_cov.save()
+
+        return ori_cov
+
+    @classmethod
+    def update(cls, element_id, **kwargs):
+        ori_cov = OriginalCoverage.objects.get(id=element_id)
+        if ori_cov:
+            # # update projection string info
+            for key, value in kwargs.iteritems():
+                if key in ('projection_string','projection_name','datum','unit', 'northlimit', 'southlimit', 'eastlimit', 'westlimit'):
+                    setattr(ori_cov, key, value)
+                    ori_cov.save()
+        else:
+            raise ObjectDoesNotExist("No coverage element was found for the provided id: %d" % element_id)
+
+    @classmethod
+    def remove(cls, element_id):
+        ori_cov = OriginalCoverage.objects.get(id=element_id)
+        if ori_cov:
+            ori_cov.delete()
+        else:
+            raise ObjectDoesNotExist("No original coverage element exists for id: %d."%element_id)
```

Comment:
Again, all of this is terribly redundant. It looks like its getting copy-pasted and minimal edits for every new metadata element? Is that what's happening?